src/components/select/simple-select/simple-select.test.js

```diff
@@ -0,0 +1,1022 @@
+            listPlacement={position}
+            mt={top}
+            mb={bottom}
+            ml={left}
+            mr={right}
+          />
+        );
+
+        selectText().click();
+        commonDataElementInputPreview().should(
+          "have.attr",
+          "aria-expanded",
+          "true"
+        );
+        selectListPosition()
+          .should("have.attr", "data-popper-placement", position)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      ["top", "0px", "0px", "0px", "0px"],
+      ["bottom", "600px", "0px", "0px", "0px"],
+      ["left", "200px", "0px", "0px", "900px"],
+      ["right", "200px", "0px", "500px", "0px"],
+    ])(
+      "should flip list to opposite position when there is not enough space to render it in %s position",
+      (position, top, bottom, left, right) => {
+        CypressMountWithProviders(
+          <SimpleSelectComponent
+            listPlacement={position}
+            flipEnabled
+            mt={top}
+            mb={bottom}
+            ml={left}
+            mr={right}
+          />
+        );
+
+        let flipPosition;
+        if (position === "top") {
+          flipPosition = "bottom";
+        }
+        if (position === "bottom") {
+          flipPosition = "top";
+        }
+        if (position === "left") {
+          flipPosition = "right";
+        }
+        if (position === "right") {
+          flipPosition = "left";
+        }
+
+        selectText().click();
+        commonDataElementInputPreview().should(
+          "have.attr",
+          "aria-expanded",
+          "true"
+        );
+        selectListPosition()
+          .should("have.attr", "data-popper-placement", flipPosition)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      [true, "not.be.visible"],
+      [false, "be.visible"],
+    ])("should check the disablePortal prop when %s", (state, visibility) => {
+      CypressMountWithProviders(
+        <Dialog open height="60px">
+          <SimpleSelectComponent disablePortal={state} />
+        </Dialog>
+      );
+
+      selectText().click();
+      selectList().should(visibility);
+    });
+  });
+
+  describe("check events for Simple Select component", () => {
+    it("should return onChange event when a list option is selected", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(
+        <SimpleSelectEventsComponent onChange={callback} />
+      );
+
+      const position = "first";
+
+      selectText().click();
+      selectOption(positionOfElement(position))
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should return onBlur event when the list is closed with the Tab key", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<SimpleSelectComponent onBlur={callback} />);
+
+      selectText().click();
```

Comment:
we can simplify this, but still need to an assertion for checking `onBlur` callback

```suggestion
      selectText().click().tab();
```